library/Icingadb/Web/Controller.php

```diff
@@ -154,7 +154,7 @@ public function createSortControl(Query $query, array $columns)
     public function createSearchBar(Query $query, array $preserveParams = null)
     {
         $requestUrl = Url::fromRequest();
-        $redirectUrl = $preserveParams !== null ? $requestUrl->onlyWith($preserveParams) : $requestUrl;
+        $redirectUrl = $preserveParams !== null ? $requestUrl->onlyWith($preserveParams) : $requestUrl->setParams([]);
```

Comment:
Way better already :smile: But `onlyWith` clones the url first, `setParams` does not. But the cloning should happen because changing the request url implicitly isn't good.